app/controllers/application_controller.rb

```diff
@@ -70,6 +70,15 @@ def current_user
     @current_user ||= User.find(session[:user_id])
   end
 
+  ###
+  # @description: Returns a pundit user object
+  ###
+  def pundit_user
+    UserContext.new(current_user, organization: current_organization,
```

Comment:
`ConfigurationProfileUser` has references to both of the other, configuration profile and organization. Can the context be initalized with either one only, without the user themselves? If not, we can only pass one argument and delegate to it to retrieve the rest.

Reply:
Good catch, will update to use `current_configuration_profile_user` as `pundit_user` at one of the next PRs

Reply:
@excelsior checked the code and we do need both - `current_user` and `current_configuration_profile_user` because sometimes `current_configuration_profile_user` will be blank. And prefer to keep other attributes too if you don't mind, will add memoization to those controller methods.

Reply:
Ah, yes, we absolutely need `current_user`. I'm just doubting the need of passing organization and profile alongside profile user when instantiating `UserContext`.

Reply:
Will keep configuration_profile and configuration_profile_user as configuration_profile will be queried anyway for building configuration_profile_user